app/src/pages/nanomine/tools/MCRToolTemplate/MCRToolTemplate.js

```diff
@@ -0,0 +1,296 @@
+import ReferenceContainer from '@/components/nanomine/ReferenceContainer'
+import Dialog from '@/components/Dialog'
+import ImageUpload from '@/components/nanomine/ImageUpload'
+import { JobMgr } from '@/modules/JobMgr.js'
+import ToolCard from '@/components/nanomine/ToolCard'
+import Spinner from '@/components/Spinner.vue'
+
+import { mapGetters, mapMutations } from 'vuex'
+// import {Auth} from '@/modules/Auth.js'
+import Jszip from 'jszip'
+
+export default {
+  name: 'MCRToolTemplate',
+  components: {
+    ToolCard,
+    ReferenceContainer,
+    ImageUpload,
+    dialogBox: Dialog,
+    Spinner
+  },
+  props: {
+    card: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    name: {
+      type: String,
+      required: true
+    },
+    job: {
+      type: Object,
+      required: false
+    },
+    header: {
+      type: String,
+      required: false
+    },
+    referenceList: {
+      type: Array,
+      required: false,
+      default: () => []
+    }
+  },
+  sockets: {
+    finished: function (data) {
+      this.results.jobid = data
+      this.results.uri = '/nmf/jobdata/' + data
+      this.setLoading()
+      return fetch(this.results.uri + '/job_output_parameters.json')
+        .then(function (response) {
+          this.results.files = response.data.files // use files array instead of individual file references
+          this.results.obtained = true
+          this.resetLoading()
+        })
+        .catch(function (err) {
+          this.renderDialog({
+            title: 'Socket Error',
+            content: err.message,
+            reason: 'socketError'
+          })
+          this.resetLoading()
+        })
+    },
+    hello: function (data) {
+      if (data === 'connection received' && this.job?.useWebsocket === true) {
+        this.useWebsocket = true
+      }
+    }
+  },
+  data: function () {
+    return {
+      jobId: '',
+      dialog: {
+        title: '',
+        content: '',
+        reason: ''
+      },
+      files: undefined,
+      selectedOptions: {},
+      useWebsocket: false,
+      results: {
+        obtained: false,
+        files: undefined,
+        uri: undefined,
+        jobid: undefined,
+        submitted: false,
+        downloading: false
+      },
+      auth: {
+        // AUTH MOCKED because auth is not yet implemented
+        isLoggedIn: () => false,
+        isTestUser: () => false
+      },
+      cardSlots: ['image', 'title', 'content', 'actions']
+    }
+  },
+  beforeMount: function () {
+    // this.auth = new Auth()
+    if (!this.auth.isLoggedIn()) {
+      this.renderDialog({
+        title: 'Authorization Error',
+        content: 'Login is required, please log in',
+        reason: 'loginRequired'
+      })
+    }
+  },
+  mounted () {
+    this.resetContent()
+    this.$socket.emit('testConnection')
```

Comment:
Handling WebSocket interactions inside a Vuex action is what I'll suggest here. I'll propose a TD ticket to fix this